File: src/main/java/com/github/repository/SessionResource.java
package com.github.repository;

import com.github.utils.M;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import java.util.Map;

/**
 * Created by zk_chs on 16/4/11.
 */
@RestController
@RequestMapping(value = "/session")
public class SessionResource {

    @RequestMapping(value = "/set", method = {RequestMethod.GET, RequestMethod.POST})
    public Map<String, Object> set (HttpServletRequest request){
        request.getSession().setAttribute("request Url", request.getRequestURL());
        return M.builder()
                .put("result", "success")
                .put("request Url", request.getRequestURL())
                .buildMap();
    }

    @RequestMapping(value = "/get", method = {RequestMethod.GET, RequestMethod.POST})
    public Map<String, Object> get (HttpServletRequest request){
        return M.builder()
                .put("sessionId", request.getSession().getId())
                .put("message", request.getSession().getAttribute("request Url"))
                .buildMap();
    }

}

File: src/main/scala/com/github/resources/SessionResource.scala
package com.github.resources

import java.util
import javax.servlet.http.HttpServletRequest

import com.github.utils.M
import org.springframework.web.bind.annotation.{RequestMethod, RequestMapping, RestController}

import scala.language.postfixOps

/**
  * Created by zk_chs on 16/4/13.
  */
@RestController
@RequestMapping(value = Array("/session"))
class SessionResource {

  @RequestMapping(value = Array("/set"), method = Array(RequestMethod GET, RequestMethod POST))
  def set(request: HttpServletRequest) = {
    request getSession() setAttribute ("session_msg", request getRequestURL)
    M builder() put("result", "success") put ("session_id", request getSession() getId) put ("session_msg", request getRequestURL) buildMap
  }

  @RequestMapping(value = Array("/get"), method = Array(RequestMethod GET, RequestMethod POST))
  def get(request: HttpServletRequest) =  {
    M builder() put ("result", "success") put ("session_id", request getSession() getId) put ("session_msg", request getSession() getAttribute "session_msg") buildMap
  }

}

class Animal
class Bird extends Animal {
  val animalList = List(new Animal, new Animal)
  new Bird :: animalList
  :: (new Bird, animalList)
}
